service/src/main/java/org/apache/griffin/core/job/JobServiceImpl.java

```diff
@@ -288,6 +288,11 @@ public void deleteJob(String name) throws SchedulerException {
         return updateState(instances);
     }
 
+    @Override
+    public JobInstanceBean findInstance(Long id) {
+        return instanceRepo.findByInstanceId(id);
+    }
```

Comment:
In case there is no bean with given id, I think, it would be better to throw an exception with explanation. We can log this information as well.

Reply:
what will happen if it's left as-is?

Reply:
I think, it will return null or an exception without any information, what exactly happens.

Reply:
done